BE build break

<!-- (At least include the following, feel free to add if you have more content) -->

### Steps to reproduce the behavior (Required)

```
[1/3] Building CXX object src/exprs/CMakeFiles/Exprs.dir/agg/aggregate_factory.cpp.o
FAILED: src/exprs/CMakeFiles/Exprs.dir/agg/aggregate_factory.cpp.o 
ccache /home/disk1/sr-deps/toolchain/installed/gcc-10.3.0/bin/g++ -DHAVE_INTTYPES_H -DHAVE_NETINET_IN_H -I/home/disk1/caixiaohua/git-src/starrocks/be/src/formats/orc/apache-orc/c++/include -I/home/disk1/caixiaohua/git-src/starrocks/thirdparty/installed/include -I/home/disk1/caixiaohua/starlet-installed/third_party/include -I/home/disk1/caixiaohua/starlet-installed/starlet_install/include -I/home/disk1/caixiaohua/git-src/starrocks/be/src -I/home/disk1/caixiaohua/git-src/starrocks/be/test -I/home/disk1/caixiaohua/git-src/starrocks/be/../gensrc/build -I/home/disk1/caixiaohua/git-src/starrocks/thirdparty/installed/gperftools/include -I/home/disk1/caixiaohua/git-src/starrocks/thirdparty/installed/include/thrift -I/home/disk1/caixiaohua/git-src/starrocks/thirdparty/installed/include/event -I/home/disk1/caixiaohua/git-src/starrocks/thirdparty/installed/include/breakpad -I/home/disk1/caixiaohua/git-src/starrocks/thirdparty/installed/open_jdk/include -I/home/disk1/caixiaohua/git-src/starrocks/thirdparty/installed/open_jdk/include/linux -Wall -Wno-sign-compare -Wno-unknown-pragmas -pthread -Wno-register -Wno-strict-aliasing -fno-omit-frame-pointer -std=gnu++17 -D__STDC_FORMAT_MACROS -Wno-deprecated -Wno-vla -Wno-comment -DBOOST_DATE_TIME_POSIX_TIME_STD_CONFIG -DBOOST_SYSTEM_NO_DEPRECATED -DBOOST_UUID_RANDOM_PROVIDER_FORCE_POSIX -Werror=return-type -Werror=switch -DUSE_STAROS -msse4.2 -mavx2  -Wno-attributes -DS2_USE_GFLAGS -DS2_USE_GLOG -faligned-new -gz=zlib  -Werror -g -Wno-unused-local-typedefs -O3 -gdwarf-4 -DNDEBUG -O3 -DNDEBUG -MD -MT src/exprs/CMakeFiles/Exprs.dir/agg/aggregate_factory.cpp.o -MF src/exprs/CMakeFiles/Exprs.dir/agg/aggregate_factory.cpp.o.d -o src/exprs/CMakeFiles/Exprs.dir/agg/aggregate_factory.cpp.o -c /home/disk1/caixiaohua/git-src/starrocks/be/src/exprs/agg/aggregate_factory.cpp
In file included from /home/disk1/caixiaohua/git-src/starrocks/be/src/exprs/agg/aggregate_factory.cpp:27:
/home/disk1/caixiaohua/git-src/starrocks/be/src/exprs/agg/percentile_cont.h: In member function ‘void starrocks::vectorized::PercentileContAggregateFunction<PT>::finalize_to_column(starrocks_udf::FunctionContext*, starrocks::vectorized::ConstAggDataPtr, starrocks::vectorized::Column*) const [with starrocks::PrimitiveType PT = starrocks::TYPE_DECIMAL64]’:
/home/disk1/caixiaohua/git-src/starrocks/be/src/exprs/agg/percentile_cont.h:111:20: error: ‘result’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
  111 |         ResultType result;
      |                    ^~~~~~
/home/disk1/caixiaohua/git-src/starrocks/be/src/exprs/agg/percentile_cont.h: In member function ‘void starrocks::vectorized::AggregateFunctionBatchHelper<State, Derived>::batch_finalize(starrocks_udf::FunctionContext*, size_t, starrocks::vectorized::Buffer<unsigned char*>&, size_t, starrocks::vectorized::Column*) const [with State = starrocks::vectorized::PercentileContState<starrocks::TYPE_DECIMAL64, int>; Derived = starrocks::vectorized::PercentileContAggregateFunction<starrocks::TYPE_DECIMAL64>]’:
/home/disk1/caixiaohua/git-src/starrocks/be/src/exprs/agg/percentile_cont.h:111:20: error: ‘result’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
/home/disk1/caixiaohua/git-src/starrocks/be/src/exprs/agg/percentile_cont.h: In member function ‘void starrocks::vectorized::PercentileContAggregateFunction<PT>::finalize_to_column(starrocks_udf::FunctionContext*, starrocks::vectorized::ConstAggDataPtr, starrocks::vectorized::Column*) const [with starrocks::PrimitiveType PT = starrocks::TYPE_DECIMAL32]’:
/home/disk1/caixiaohua/git-src/starrocks/be/src/exprs/agg/percentile_cont.h:111:20: error: ‘result’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
/home/disk1/caixiaohua/git-src/starrocks/be/src/exprs/agg/percentile_cont.h: In member function ‘void starrocks::vectorized::AggregateFunctionBatchHelper<State, Derived>::batch_finalize(starrocks_udf::FunctionContext*, size_t, starrocks::vectorized::Buffer<unsigned char*>&, size_t, starrocks::vectorized::Column*) const [with State = starrocks::vectorized::PercentileContState<starrocks::TYPE_DECIMAL32, int>; Derived = starrocks::vectorized::PercentileContAggregateFunction<starrocks::TYPE_DECIMAL32>]’:
/home/disk1/caixiaohua/git-src/starrocks/be/src/exprs/agg/percentile_cont.h:111:20: error: ‘result’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
/home/disk1/caixiaohua/git-src/starrocks/be/src/exprs/agg/percentile_cont.h: In member function ‘void starrocks::vectorized::PercentileContAggregateFunction<PT>::finalize_to_column(starrocks_udf::FunctionContext*, starrocks::vectorized::ConstAggDataPtr, starrocks::vectorized::Column*) const [with starrocks::PrimitiveType PT = starrocks::TYPE_DECIMAL128]’:
/home/disk1/caixiaohua/git-src/starrocks/be/src/exprs/agg/percentile_cont.h:111:20: error: ‘result’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
/home/disk1/caixiaohua/git-src/starrocks/be/src/exprs/agg/percentile_cont.h: In member function ‘void starrocks::vectorized::AggregateFunctionBatchHelper<State, Derived>::batch_finalize(starrocks_udf::FunctionContext*, size_t, starrocks::vectorized::Buffer<unsigned char*>&, size_t, starrocks::vectorized::Column*) const [with State = starrocks::vectorized::PercentileContState<starrocks::TYPE_DECIMAL128, int>; Derived = starrocks::vectorized::PercentileContAggregateFunction<starrocks::TYPE_DECIMAL128>]’:
/home/disk1/caixiaohua/git-src/starrocks/be/src/exprs/agg/percentile_cont.h:111:20: error: ‘result’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
cc1plus: all warnings being treated as errors
ninja: build stopped: subcommand failed.
```

### Expected behavior (Required)

No Build Break

### Real behavior (Required)

Build fails due to an warning as error

### StarRocks version (Required)

current `main` branch, introduced by PR #7737

--- a/be/src/exprs/agg/percentile_cont.h
+++ b/be/src/exprs/agg/percentile_cont.h
@@ -108,7 +108,7 @@ public:
         double u = (new_vector.size() - 1) * rate;
         int index = (int)u;
 
-        ResultType result;
+        [[maybe_unused]] ResultType result;
         if constexpr (pt_is_datetime<PT>) {
             result.from_unix_second(
                     new_vector[index].to_unix_second() +

--- a/be/src/exprs/agg/percentile_cont.h
+++ b/be/src/exprs/agg/percentile_cont.h
@@ -120,6 +120,7 @@ public:
             result = new_vector[index] + (u - (double)index) * (new_vector[index + 1] - new_vector[index]);
         } else {
             LOG(ERROR) << "Invalid PrimitiveTypes for percentile_cont function";
+            return;
         }
 
         column->append(result);

